referenceframe/urdf.go

```diff
@@ -0,0 +1,258 @@
+package referenceframe
+
+import (
+	"encoding/xml"
+	"fmt"
+	"math"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/golang/geo/r3"
+	"github.com/pkg/errors"
+
+	spatial "go.viam.com/rdk/spatialmath"
+)
+
+// URDFConfig represents all supported fields in a Universal Robot Description Format (URDF) file.
+type URDFConfig struct {
+	XMLName xml.Name `xml:"robot"`
+	Name    string   `xml:"name,attr"`
+	Links   []struct{
+		XMLName   xml.Name `xml:"link"`
+		Name      string   `xml:"name,attr"`
+		Collision []struct{
+			XMLName xml.Name `xml:"collision"`
+			Name    string   `xml:"name,attr"`
+			Origin  struct{
+				XMLName xml.Name `xml:"origin"`
+				RPY     string   `xml:"rpy,attr"`  // "r p y" format
```

Comment:
I had tried as a []float64 and it fails to parse the XML. I suspect you need a custom decoder from my readings of the `encoding/xml` package, but I haven't investigated further yet.

Reply:
cool just was curious.  your way works nice too

Reply:
Otherwise, I'm with you. I'd like to have those just come in as the data structure we care about. Maybe something we could improve in the future when we have some time to experiment with the `xml` package.